tests/test_tpp_backend.py

```diff
@@ -103,6 +110,16 @@ def test_sql_error_propagates_with_sqlcmd():
             assert "Invalid object name 'Bar'" in str(excinfo.value)
 
 
+def test_credentials_error_propagates_with_sqlcmd(set_database_url_with_bad_password):
+    with patch.object(TPPBackend, "to_sql") as to_sql:
```

Comment:
Rather than patching, can't we just call `mssql_query_to_csv_file` directly and assert that it raised the right error? You wouldn't even need to monkeypatch DATABASE_URL in that case, as it's supplied directly as an argument.